PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java

```diff
@@ -55,6 +58,40 @@ public void configureJavaProject(String newProjectCompliance, IProgressMonitor m
 			if (parent == null) { 
 				parent = (HDInsightsScalaProjectWizard) this.getWizard();
 			}
+			
+			if (hasConfiguredScalaClasspathContainer == false) {
+				hasConfiguredScalaClasspathContainer = true;
+				Display.getDefault().syncExec(new Runnable() {
+				    public void run() {
+						IJavaProject javaProject = getJavaProject();
+						IClasspathEntry[] entries = null;
+						try {
+							entries = javaProject.getRawClasspath();
+							IClasspathEntry scalaClasspathContainerEntry = null;
+							int scalaClasspathContainerEntryIndex = -1;
+							for (int i = 0; i < entries.length; i++) {
```

Comment:
Java8 stream would be helperful here. :smile:

Reply:
In general agree. For this one, logic is tricky so prefer to keep it as is.

---
In reply to: [144226667](https://github.com/Microsoft/azure-tools-for-java/pull/1098#discussion_r144226667) [](ancestors = 144226667)